libs/storage/Tsavorite/cs/src/core/VarLen/SpanByteExtensions.cs

```diff
@@ -13,85 +13,49 @@ public static unsafe class SpanByteExtensions
         /// <summary>
         /// Upsert with Span input
         /// </summary>
-        /// <param name="clientSession"></param>
-        /// <param name="key"></param>
-        /// <param name="desiredValue"></param>
-        /// <param name="userContext"></param>
-        /// <param name="serialNo"></param>
-        /// <returns></returns>
         public static Status Upsert<Input, Output, Context, Functions>(this ClientSession<SpanByte, SpanByte, Input, Output, Context, Functions> clientSession, ReadOnlySpan<byte> key, ReadOnlySpan<byte> desiredValue, Context userContext = default, long serialNo = 0)
             where Functions : IFunctions<SpanByte, SpanByte, Input, Output, Context>
         {
-            fixed (byte* k = key)
-            fixed (byte* v = desiredValue)
-                return clientSession.Upsert(SpanByte.FromFixedSpan(key), SpanByte.FromFixedSpan(desiredValue), userContext, serialNo);
+            fixed (byte* keyPtr = key)
+            fixed (byte* desiredValuePtr = desiredValue)
+                return clientSession.Upsert(SpanByte.FromPointer(keyPtr, key.Length), SpanByte.FromPointer(desiredValuePtr, desiredValue.Length), userContext, serialNo);
```

Comment:
Similar as my previous comment, this loses the ``Fixed`` hint and it is very important the ptr is fixed.

Reply:
I'll add "Pinned" word to this overload to make it clearer, good catch.